### PR TITLE
tag alias

### PR DESCRIPTION
**Pull request by brianteeman**
You should be able to create tags with parents and children such as

- Sport
-- adults
-- children
- games
-- adults
-- children

Before this PR you could not as you would get a duplicate alias message error even though the tags have different parents and there is no reason that I can see to prevent this *_**I may be very wrong as I dont use tags that often**_*

Pull Request for Issue #21326 .

### CODE REVIEW

**Comment by Franzwohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 46f47fb455bb4b8db6590c2e58383643083d6d94<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38576">issues.joomla.org/tracker/joomla-cms/38576</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 46f47fb455bb4b8db6590c2e58383643083d6d94<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38576">issues.joomla.org/tracker/joomla-cms/38576</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38576">issues.joomla.org/tracker/joomla-cms/38576</a>.</sub>

**Comment by fancyFranci** [comment]
To be sure that there are no side effects we do not see, please move it to 4.3. Especially with your new tag feature this fix will have the chance to receive deeper tests.

**Comment by brianteeman** [comment]
>  please move it to 4.3.

You already did ;(

**Comment by obuisard** [comment]
Thank you Brian @brianteeman

File: libraries/src/Table/ContentType.php [hunk 1]
```diff
@@ -81,7 +81,10 @@ public function store($updateNulls = false)
         // Verify that the alias is unique
         $table = Table::getInstance('Contenttype', 'JTable', array('dbo' => $this->getDbo()));
 
-        if ($table->load(array('type_alias' => $this->type_alias)) && ($table->type_id != $this->type_id || $this->type_id == 0)) {
+        if (
+            $table->load(array('alias' => $this->alias, 'parent_id' => (int) $this->parent_id))
+            && ($table->id != $this->id || $this->id == 0)
+        ) {
```
Comment (heelc29) [inline libraries/src/Table/ContentType.php:87]: @brianteeman Are you sure this change is correct?. The `alias` and `parent_id` fields do not exist in `#__content_types` table and I think using the lang key `COM_TAGS_ERROR_UNIQUE_ALIAS` does not belong here and was wrong even before this PR.

Comment (hikashop-nicolas) [reply libraries/src/Table/ContentType.php:87]: I'm currently testing the Joomla 4.3 Beta 4 with HikaShop. During the installation process, we use this store function to add the tags capability on the products from HikaShop. We get the error:
Missing field in database: Joomla\CMS\Table\ContentType &#160; alias.
https://i.imgur.com/yRGq5wK.png
Reverting this change does fix the problem. But I'm not familiar enough with the tag system to know if the code change is ok and we should change something in HikaShop (but what ?) or if another kind of modification should be necessary in Joomla 4.3 to fix the tag alias issue while preventing this error.

Comment (brianteeman) [reply libraries/src/Table/ContentType.php:87]: yes this PR is wrong

**Pull request merged**